fix(linter): false negative in no-restriced-imports with `patterns` and side effects

camc314 opened the pull request:
previously, when using `patterns` with side effects imports, the diagnostic would not be reported, this PR fixes it by checking the module name against patterns.

fixes #10984

Boshen commented:
### Merge activity

* **May 14, 10:03 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **May 14, 10:03 AM EDT**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

camc314 committed:
fix(linter): false negative in no-restriced-imports with `patterns` and side effects (#11027)

previously, when using `patterns` with side effects imports, the diagnostic would not be reported, this PR fixes it by checking the module name against patterns.

fixes #10984
